Include/object.h

```diff
@@ -279,6 +279,13 @@ _Py_ThreadId(void)
     __asm__ ("" : "=r" (tp));
     tid = tp;
     #endif
+#elif defined(__s390__)
+    #if defined(__clang__) && _Py__has_builtin(__builtin_thread_pointer)
+    tid = (uintptr_t)__builtin_thread_pointer();
```

Comment:
I think we should use `__builtin_thread_pointer()` on for s390 on both GCC and Clang. Glibc has used it since 2.3.2 (20+ years). We should not guard it with `_Py__has_builtin` or `__has_builtin` because the `__builtin_thread_pointer()` support is older than the `__has_builtin` support.